python/lsst/daf/fmt/mysql/mysqlStorage.py

```diff
@@ -0,0 +1,448 @@
+        If False then a new repository will not be created.
+
+    Raises
+    ------
+    NoRepositoryAtRoot
+        If create is False and a repository does not exist at the root
+        specified by uri then NoRepositoryAtRoot is raised.
+    """
+
+    @staticmethod
+    def registerFormatter(formattable, formatter):
+        if formattable in MySqlStorage.formatters:
+            raise RuntimeError("Registration of second formatter {} for " +
+                               "formattable class {}".format(formatter,
+                                                             formattable))
+        MySqlStorage.formatters[formattable] = formatter
+
+    formatters = {}
+
+    def __init__(self, uri, create):
+        self.mysql = MySQLdb.connect(read_default_file="~/.my.cnf")
+        self.uri = uri
+        parseRes = urlparse.urlparse(uri)
+        try:
+            self.host, self.port = parseRes.netloc.split(':')
+            self.port = int(self.port)
+        except ValueError:
+            raise RuntimeError(
+                "Could not interpret {} to get host and port.".format(
+                    parseRes.netloc))
+        self.repoDbName = parseRes.path.strip('/')
+
+        if create:
+            self._createDb()
+        else:
+            if self._repoDbExists() is False:
+                raise NoRepositoryAtRoot(
+                    "Database {} does not exist.".format(self.repoDbName))
+
+    def identifierQuote(self, field):
+        """Wrap a field identifier in identifier quote characters. Which quote
+        character is used can depend on the state of the backing database (but
+        this is not yet implemented).
+
+        Parameters
+        ----------
+        field : string
+            A field identifier to be wrapped in identifier quote characters.
+
+        Returns
+        -------
+        string
+            The field identifier wrapped in identifier quote characters.
+        """
+        if '`' in field:
+            raise RuntimeError('Illegal ` character in field.')
+        return "`{}`".format(field)
+
+    def _connection(self):
+        return MySQLdb.connect(host=self.host, port=self.port,
+                               db=self.repoDbName,
+                               read_default_file="~/.my.cnf")
+
+    def _createDb(self):
+        self._testSqlIsClean(self.repoDbName)
+        # Ideally the execute should be executing the string format (execute
+        # would be called with 2 input arguments; the string and the
+        # subtitution values), but I can't figure out how to make it work on
+        # CREATE DATABASE
+        self.mysql.cursor().execute("CREATE DATABASE IF NOT EXISTS %s" %
+                                    self.repoDbName)
+
+    def _repoDbExists(self):
+        sql = """SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE
+            SCHEMA_NAME = %s;"""
+        cursor = self.mysql.cursor()
+        len = cursor.execute(sql, (self.repoDbName,))
```

Comment:
This is not very portable, DBAPI does not specify what `cursor.execute()` should return.